### PR TITLE
shouldNotGet function added for the CouchbaseSystem

### PR DESCRIPTION
**Pull request by AksalBilal**

### CODE REVIEW

File: lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt [hunk 1]
```diff
@@ -95,6 +98,21 @@ class CouchbaseSystem internal constructor(
         .let(assertion)
         .let { this }
 
+    override suspend fun shouldNotGet(
```
Comment (Bramix) [inline lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt:101]: shouldNotGet naming looks not so clear for me.
Mb "ShouldNotExistById" or "ShouldNotExist" will be better

File: lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt
Comment (Bramix) [inline lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt:1]: I think we can also add this functionality for all other document databases. Like mongo, elastic. 
What do you think?

File: lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt (hunk 1 above)
Comment (AksalBilal) [reply lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt:101]: I changed to ShouldNotExist

File: lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt [hunk 2]
```diff
@@ -95,6 +98,21 @@ class CouchbaseSystem internal constructor(
...
+        key: String,
+    ): CouchbaseSystem = when (
+        collection.get(key)
+            .onErrorResume { throwable ->
+                when (throwable) {
+                    is DocumentNotFoundException -> Mono.empty()
```
[the remark this answers is not in the document]
Comment (osoykan) [reply lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt:107]: Seems like a candidate for an extension. You can easily `foreach` at the consumer side; I don't think we should maintain that behavior in the framework itself.

```kotlin
(1...10).forEach{
  shouldNotExists(it)
}
```

File: lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt
Comment (AksalBilal) [reply lib/stove-testing-e2e-couchbase/src/main/kotlin/com/trendyol/stove/testing/e2e/couchbase/CouchbaseSystem.kt:1]: Yes, I will add this implementation also for the other existing databases

File: lib/stove-testing-e2e-mongodb/src/test/kotlin/com/trendyol/stove/testing/e2e/mongodb/MongodbTestSystemTests.kt [hunk 3]
```diff
@@ -96,84 +98,97 @@ class MongodbTestSystemTests : FunSpec({
...
+                )
+                shouldGet<ExampleInstanceWithStringObjectId>(id1.toHexString()) { actual ->
+                    actual.aggregateId shouldBe id1.toHexString()
+                }
+                assertThrows<AssertionError> { shouldNotExist(id1.toHexString()) }
+            }
+        }
+    }
+
+    test("should does not throw exception when given does not exist id") {
```
Comment (osoykan) [inline lib/stove-testing-e2e-mongodb/src/test/kotlin/com/trendyol/stove/testing/e2e/mongodb/MongodbTestSystemTests.kt:166]: typo: remove `does`

**Comment by osoykan** [comment]
|File|Coverage [74.65%]|:green_apple:|
|:-|:-:|:-:|
|[MongodbSystem.kt](https://github.com/Trendyol/stove4k/blob/5d071e30f597d8f54960bea07f3b295c6502875a/lib%2Fstove-testing-e2e-mongodb%2Fsrc%2Fmain%2Fkotlin%2Fcom%2Ftrendyol%2Fstove%2Ftesting%2Fe2e%2Fmongodb%2FMongodbSystem.kt)|92.87%|:green_apple:|
|[CouchbaseSystem.kt](https://github.com/Trendyol/stove4k/blob/5d071e30f597d8f54960bea07f3b295c6502875a/lib%2Fstove-testing-e2e-couchbase%2Fsrc%2Fmain%2Fkotlin%2Fcom%2Ftrendyol%2Fstove%2Ftesting%2Fe2e%2Fcouchbase%2FCouchbaseSystem.kt)|69.54%|:green_apple:|
|[ElasticsearchSystem.kt](https://github.com/Trendyol/stove4k/blob/5d071e30f597d8f54960bea07f3b295c6502875a/lib%2Fstove-testing-e2e-elasticsearch%2Fsrc%2Fmain%2Fkotlin%2Fcom%2Ftrendyol%2Fstove%2Ftesting%2Fe2e%2Felasticsearch%2FElasticsearchSystem.kt)|65.61%|:green_apple:|
|[DocumentDatabaseSystem.kt](https://github.com/Trendyol/stove4k/blob/5d071e30f597d8f54960bea07f3b295c6502875a/lib%2Fstove-testing-e2e%2Fsrc%2Fmain%2Fkotlin%2Fcom%2Ftrendyol%2Fstove%2Ftesting%2Fe2e%2Fdatabase%2FDocumentDatabaseSystem.kt)|25%|:x:|

|Total Project Coverage|63.79%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**